File: python/algorithms/SelectionSort/selection_sort.py
def asc_selection_sort(arr):
    length = len(arr)
    i = 0
    while (i < length - 1):
        min = i
        j = i + 1
        while (j < length):
            if (arr[min] > arr[j]):
                min = j
            j += 1
        if (i != min):
            arr[i], arr[min] = arr[min], arr[i]
        i += 1

    return arr


def des_selection_sort(arr):
    length = len(arr)
    i = 0
    while (i < length - 1):
        max = i
        j = i + 1
        while (j < length):
            if (arr[max] < arr[j]):
                max = j
            j += 1
        if (i != max):
            arr[i], arr[max] = arr[max], arr[i]
        i += 1

    return arr


if __name__ == '__main__':
    main()

File: c/Algorithms/Sort/SelectionSort/selection_sort.c
/**********************************************
 > File Name: selection_sort.c
 > Author: Signal Yu
 > Mail: signalfish38861721@gmail.com
 > Time: 11/12/2022
 > Desc:
***********************************************/

#include "selection_sort.h"
#include <stdio.h>

void selection_sort(int *arr, int size) {
  for (int i = 0; i < size - 1; i++) {
    int min = i;
    for (int j = i + 1; j < size; j++) {
      if (arr[j] < arr[min]) {
        min = j;
      }
    }
    if (i != min) {
      int temp = arr[i];
      arr[i] = arr[min];
      arr[min] = temp;
    }
  }
}